tests/unit/test_pipeline.py

```diff
@@ -93,3 +95,150 @@ def test_migrate():
     p = Pipeline("tests/data/pipeline", "pipeline-one")
     migrations = p.migrations()
     assert migrations == {"field-one": "FieldOne"}
+
+
+def test_lookups_get_max_entity_success():
+    """
+    test entity num generation functionality
+    :return:
+    """
+
+    pipeline_name = "ancient-woodland"
+    lookups = Lookups("")
+    max_entity_num = lookups.get_max_entity(pipeline_name)
+
+    assert max_entity_num == 0
+
+    entry = {
+        "prefix": "ancient-woodland",
+        "resource": "",
+        "organisation": "government-organisation:D1342",
+        "reference": "1",
+        "entity": "12344",
+    }
+    lookups.entries.append(entry)
+    expected_entity_num = 12344
+
+    assert lookups.get_max_entity(pipeline_name) == expected_entity_num
+
+    max_entity_num = lookups.get_max_entity(pipeline_name)
+    lookups.entity_num_gen.state["current"] = max_entity_num
+    lookups.entity_num_gen.state["range_max"] = max_entity_num + 10
+    expected_entity_num = 12345
+
+    assert lookups.entity_num_gen.next() == expected_entity_num
+
+
+def test_lookups_validate_entry_success():
+    """
+    test csv validate_entry functionality
```

Comment:
In all these descriptions it mentions a csv I think this is misleading so may be worth removing the word csv.